src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/oidc/OpenIDConfigBean.java

```diff
@@ -0,0 +1,65 @@
+package edu.harvard.iq.dataverse.authorization.providers.oauth2.oidc;
+
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.settings.JvmSettings;
+import edu.harvard.iq.dataverse.util.SystemConfig;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import jakarta.servlet.http.HttpServletRequest;
+
+@Named("openIdConfigBean")
+public class OpenIDConfigBean implements java.io.Serializable {
+    @Inject
+    HttpServletRequest request;
+
+    @Inject
+    AuthenticationServiceBean authenticationSvc;
+
+    public String getProviderURI() {
+        final String oidcp = request.getParameter("oidcp");
+        if (oidcp == null || oidcp == "") {
+            return JvmSettings.OIDC_AUTH_SERVER_URL.lookupOptional().orElse(null);
+        }
+        try {
+            return ((OIDCAuthProvider) authenticationSvc.getAuthenticationProvider(oidcp)).getIssuerEndpointURL();
+        } catch (Exception e) {
+            return "";
+        }
+    }
+
+    public String getClientId() {
+        final String oidcp = request.getParameter("oidcp");
+        if (oidcp == null || oidcp == "") {
+            return JvmSettings.OIDC_CLIENT_ID.lookupOptional().orElse(null);
+        }
+        try {
+            return ((OIDCAuthProvider) authenticationSvc.getAuthenticationProvider(oidcp)).getClientId();
+        } catch (Exception e) {
+            return "";
+        }
+    }
+
+    public String getClientSecret() {
+        final String oidcp = request.getParameter("oidcp");
+        if (oidcp == null || oidcp == "") {
+            return JvmSettings.OIDC_CLIENT_SECRET.lookupOptional().orElse(null);
+        }
+        try {
+            return ((OIDCAuthProvider) authenticationSvc.getAuthenticationProvider(oidcp)).getClientSecret();
+        } catch (Exception e) {
+            return "";
+        }
+    }
+
+    public String getRedirectURI() {
```

Comment:
This method is problematic.

When would someone *not* send the target? The `target` parameter is always coming from the `/oidc/login` endpoint you created. Why would anyone using the API use it? Why would there be a redirect when using the API at all? Isn't defaulting to an "API" target diverging from the "JSF" default in the `/oidc/callback/<target>` endpoint?

I do get that you need routing between JSF and SPA. But this is tricky: people already using OIDC have their IdP configured of allowing a redirect URI `<siteurl>/oauth2/callback.xhtml`. Now they need to change this setup, making their IdP use `<siteurl>/oidc/callback/JSF` as well as `<siteurl>/oidc/callback/SPA` (again, I don't ever see the API use case happen in real life).

Maybe we can make people change their setup (this is a breaking change and needs very good release notes). It certainly would help if we figure out ourselves where to send the user when the token comes back.

Coming at this from a different angle: why do we need the SPA part? The SPA is supposed to be getting a token from the IdP and hit the API (and not JSF). The SPA being a public client will require a second OIDC client registration anyway (using PKCE and all) and has its own callback URL anyway. I suppose you have in mind that the SPA then trades in the token for a session via the `/oidc/session` endpoint, eliminating the need to send the token everytime because of Session Auth. (With all flaws coming with that, but maybe OK for now.) So why would we need routing for this? Which make me circle back to the `/oauth2/callback.xhtml` thing - if we don't need special routing for SPA and API, why have a breaking change?

Reply:
http://localhost:8080/oidc/login